src/__swaps__/screens/Swap/providers/swap-provider.tsx

```diff
@@ -253,48 +259,27 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
     parameters: Omit<RapSwapActionParameters<typeof type>, 'gasParams' | 'gasFeeParamsBySpeed' | 'selectedGasFee'>;
   }) => {
     try {
-      const NotificationManager = IS_IOS ? NativeModules.NotificationManager : null;
-      NotificationManager?.postNotification('rapInProgress');
+      const degenMode = swapsStore.getState().degenMode;
+      const atomicSwapsEnabled =
+        (atomicSwapsExperimentalFlag || config.atomic_swaps_enabled) &&
+        (await getShouldDelegate(parameters.chainId, parameters.quote as Quote | CrosschainQuote, parameters.assetToSell));
+      const shouldDelegate = atomicSwapsEnabled && (await getShouldDelegate(parameters.chainId, parameters.quote, parameters.assetToSell));
+      const selectedGas = getSelectedGas(parameters.chainId);
 
-      const provider = getProvider({ chainId: parameters.chainId });
       const connectedToAnvil = useConnectedToAnvilStore.getState().connectedToAnvil;
+      const chainId = connectedToAnvil ? ChainId.anvil : parameters.chainId;
+      const nonce = await getNextNonce({ address: parameters.quote.from, chainId });
```

Comment:
depending on how long the getNextNonce and getShouldDelegate functions each run, you could parallelise them with `Promise.allSettled`